src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorage.cs

```diff
@@ -0,0 +1,223 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.SQLite.Interop;
+
+namespace Microsoft.CodeAnalysis.SQLite
+{
+    /// <summary>
+    /// Implementation of an <see cref="IPersistentStorage"/> backed by SQLite.
+    /// </summary>
+    internal partial class SQLitePersistentStorage : AbstractPersistentStorage
+    {
+        /// <summary>
+        /// Inside the DB we have a table dedicated to storing strings that also provides a unique 
+        /// integral ID per string.  This allows us to store data keyed in a much more efficient
+        /// manner as we can use those IDs instead of duplicating strings all over the place.  For
+        /// example, there may be many pieces of data associated with a file.  We don't want to 
+        /// key off the file path in all these places as that would cause a large amount of bloat.
```

Comment:
No.  sqlite does not do string interning.  I found this information online (thought i can't find the link), and i tested manually.  There is *huge* savings by doing this (i.e. like 10x).